lighthouse-core/audits/legacy-javascript.js

```diff
@@ -276,23 +283,46 @@ class LegacyJavascript extends Audit {
   }
 
   /**
-   * Returns a collection of match results grouped by script url and with a mapping
-   * to determine the order in which the matches were discovered.
+   * Returns a collection of match results grouped by script url.
    *
    * @param {CodePatternMatcher} matcher
    * @param {LH.GathererArtifacts['ScriptElements']} scripts
    * @param {LH.Artifacts.NetworkRequest[]} networkRecords
+   * @param {LH.Artifacts.Bundle[]} bundles
    * @return {Map<string, PatternMatchResult[]>}
    */
-  static detectCodePatternsAcrossScripts(matcher, scripts, networkRecords) {
+  static detectAcrossScripts(matcher, scripts, networkRecords, bundles) {
     /** @type {Map<string, PatternMatchResult[]>} */
     const urlToMatchResults = new Map();
+    const polyfillData = this.getPolyfillData();
 
     for (const {requestId, content} of Object.values(scripts)) {
       if (!content) continue;
       const networkRecord = networkRecords.find(record => record.requestId === requestId);
       if (!networkRecord) continue;
+
+      // Start with pattern matching against the downloaded script.
       const matches = matcher.match(content);
+
+      // If it's a bundle with source maps, add in the polyfill modules by name too.
+      const bundle = bundles.find(b => b.script.src === networkRecord.url);
+      if (bundle) {
+        for (const {module, name} of polyfillData) {
+          // Skip if the pattern matching found a match for this polyfill.
+          if (matches.some(m => m.name === name)) continue;
+
+          const source = bundle.rawMap.sources.find(source => source.includes(module));
```

Comment:
I was avoiding `endsWith` here because I know that webpack sometimes adds `?` to the end of things ... I don't know if that's actually true from a sane configuration and maybe that's not an issue.

@patrickhulce think we can just uses `endsWith` here`?

Reply:
https://github.com/GoogleChrome/lighthouse/pull/10564/commits/69d0b6b5d0e70578ec0009f39b84d8254762904e fixed many false positives

Reply:
does it need to be `endsWith` or something more like
```js
new RegExp(`${module}($|\/|\\?)`
```

perhaps? these are the file names right?

Reply:
yeah this is the entire file path, resolvable in node. I don't think those slashes are needed. `module` is the name like `es.array.fill` and the node module is named `...es.array.fill.js`.

I'm just unclear if sometimes there is a `?` in webpack maps and if I should account for that.